hyperspy/models/model1d.py

```diff
@@ -413,7 +418,10 @@ def __call__(self, non_convolved=False, onlyactive=False,
                 component for component in component_list if component.active]
 
         if self.convolved is False or non_convolved is True:
-            axis = self.axis.axis[self.channel_switches]
+            if not ignore_channel_switches:
+                axis = self.axis.axis[self.channel_switches]
+            else:
+                axis = self.axis.axis
```

Comment:
A more compact version:
```suggestion
            slice_ = slice(None) if ignore_channel_switches else self.channel_switches
            axis = self.axis.axis[slice_]
```